Formula/tidb-server.rb

```diff
@@ -0,0 +1,89 @@
+class TidbServer < Formula
+  desc "TiDB is a MySQL compatible distributed database"
+  homepage "https://www.pingcap.com/en/"
+  url "https://github.com/pingcap/tidb.git",
+      :tag      => "v2.1.5",
+      :revision => "4dc309315c845ee1e1955754325bf2fc212e8f8d"
+
+  depends_on "go" => :build
+
+  def datadir
+    var/"tidb"
+  end
+
+  def install
+    # Removing the go.sum file
+    # fixes a problem in go mod hashes
+    rm "go.sum"
+    system "make", "server"
+
+    bin.install "bin/tidb-server"
+
+    inreplace "config/config.toml.example", "/tmp/tidb", datadir
+    etc.install "config/config.toml.example" => "tidb-server.toml"
+    datadir.mkpath
+  end
+
+  def caveats
+    s = <<~EOS
+      Note: This is a developer-install of TiDB.
```

Comment:
I don't think we want that. Homebrew ships stable, non-debug configurations by default.

Reply:
To clarify: it is both stable, and non-debug.  My own estimate is that goleveldb is 5x slower than TiKV (uses RocksDB).

Why am I not using TiKV?  To use TiKV I would also need to install pd.  The [production configuration](https://pingcap.com/docs/architecture/) is also 3x pd and 3x tikv.  It is possible to do this on a local machine, but it's quite heavy.

Reply:
We would want to ship this in the most common configuration, the one that upstream ships by default

Reply:
This is the most common configuration, for a _single server_. The nuance comes because TiDB is a distributed system.  I looked at how this is done in the `cockroach` package (also a distributed database, but with a different architecture), and there is a [similar caveat](https://github.com/Homebrew/homebrew-core/blob/master/Formula/cockroach.rb#L33-L42).

Reply:
For a longer answer to the question of _why the engine is different in a single server_:

TiDB is written in golang, and RocksDB is C++.  It is possible to link against it, but there would be FFI overhead.  It also complicates development, it's much easier to be a pure-go package.

In a distributed server environment, you can think of tidb as "sql processing nodes" and TiKV as "data nodes".  TiDB communicates to TiKV over gRPC. The TiKV server is the one doing the low-level data access, and is written in Rust instead of go.  For Rust to link against a C++ package there is no FFI overhead.

So to paraphrase, in the single server mode the TiDB server does double duty as sql processing and data storage.  It is actually the default (and what the tests-suite runs on), but it is not intended for production usage, since you lose the advantage of TiDB being a distributed system.

Reply:
@fxcoudert friendly followup ping :-)  Did my above comment address your concern?